src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ClonedConstructedEETypeNode.cs

```diff
@@ -17,7 +17,7 @@ public ClonedConstructedEETypeNode(NodeFactory factory, TypeDesc type) : base(fa
         public override ISymbolNode NodeForLinkage(NodeFactory factory) => this;
 
         //
-        // A cloned type must be named differently than the type it is a clone of so the linker
+        // A cloned type must be named differently than the type it is a clone or the linker
```

Comment:
Not a native english speaker, but the previous sentence made sense. The update doesn't.

We can replace "the type it is a clone of" with "the original type" if it's more readable like that, but the "or" is not a good change.

Reply:
Yup, as a native speaker Michal's right -- the old wording or his proposed wording are both fine, but the change here is ungrammatical.